lib/Mojo/Message.pm

```diff
@@ -97,10 +97,14 @@ sub fix_headers {
   # Content-Length or Connection (unless chunked transfer encoding is used)
   my $content = $self->content;
   my $headers = $content->headers;
+  my $websocket = $headers->sec_websocket_accept;
+  my $isWebsocket = defined $websocket;
+  my $size = $self->body_size;  
   if ($content->is_multipart) { $headers->remove('Content-Length') }
   elsif ($content->is_chunked || $headers->content_length) { return $self }
   if   ($content->is_dynamic) { $headers->connection('close') }
-  else                        { $headers->content_length($self->body_size) }
+  elsif (!$isWebsocket || $size > 0) { $headers->content_length($self->body_size) }
+  else { $headers->remove('Content-Length') }
```

Comment:
Pretty sure this code won't be merged, because of the naming convention and lack of tidying. Maybe something like this would be accepted instead?

```
  if ($content->is_dynamic) {
    $headers->connection('close');
  }
  elsif (!defined $headers->sec_websocket_accept || $self->body_size > 0) {
    $headers->content_length($self->body_size);
  }
```

Reply:
looks pretty much like the code above!? I tried to keep it as similar as possible